syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.iotservice";
option java_outer_classname = "IoTServiceProto";
option objc_class_prefix = "TEMP";

package employee_service;

service EmployeeService {
  
  // Insert a value on the list
  rpc InsertValue (Value) returns (StatusReply) {}

  // Search for a value on the list
  rpc SearchValue (Value) returns (StatusReply) {}
  
  // Remove a Value from the list 
  rpc RemoveValue (Value) returns (StatusReply) {}

  // Return the entire list
  rpc ReturnList (EmptyMessage) returns (ValueList) {}

  // Sort the list Ascending
  rpc SortAscending (EmptyMessage) returns (StatusReply) {}

  // Sort the list Descending
  rpc SortDescending (EmptyMessage) returns (StatusReply) {}
  
}

message Value {
  float number = 1;
}

message StatusReply {
  string status = 1;
}

message EmptyMessage{}

message ValueList{
  repeated Value number_data = 1;
}
